jujugui/static/gui/src/app/app.js

```diff
@@ -536,7 +528,11 @@ YUI.add('juju-gui', function(Y) {
         entityPromise = this._fetchEntityFromUserState(pathState.state.user);
       }
       this.controllerAPI = this.setUpControllerAPI(
-        controllerAPI, user, password, macaroons, entityPromise);
+        controllerAPI,
+        controller.user,
```

Comment:
Instead of this, what do you think about passing the user instance as an argument and having the `setupControllerAPI` method pull out the necessary values? If you do that, then you don't need the above declaration.